### PR TITLE
chore(lily): Refactoring and Improvements post Lily v0.10.0

### PR DESCRIPTION
**Pull request by placer14**
- tons of refactoring common parts into helpers
  - include `podManagementPolicy` value in template
  - shared app template
  - shared configmap
- redis is a dependent deployment which is managed along with the rest of lily deployment resources
- reorganize config structure which puts `daemon` values separate from `cluster.notifier` and `cluster.worker` config (but using mostly the same value structure)
  - `daemon` config is exactly the same as it existed before so this change is backward compatible with old values
  - this changes the `inlineConfig` keys to be more descriptive
- update/doc new default values

part of https://github.com/filecoin-project/sentinel-infra/issues/257

### CODE REVIEW

File: charts/lily/templates/common/_helpers.tpl [hunk 1]
```diff
@@ -0,0 +1,415 @@
+{{/*
```
Comment (frrist) [inline charts/lily/templates/common/_helpers.tpl:1]: Is there anything here that needs review specifically?

File: charts/lily/templates/daemon/app.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
+---
```
Comment (frrist) [inline charts/lily/templates/daemon/app.yaml:1]: How does this differ from the `worker` and `notifier` configs?

File: charts/lily/templates/notifier/app.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        {{- else }}
+          # Required resources to run daemon lily are signficant and the below default
+          # may be insufficient. Please see https://lilium.sh/software/lily/hardware/
+          # for more information.
+          requests:
+            cpu: "8000m"
+            memory: "64Gi"
+          limits:
+            cpu: "8000m"
+            memory: "64Gi"
```
Comment (frrist) [inline charts/lily/templates/notifier/app.yaml:82]: This changes the notifier from 256Gi RAM to 64Gi ram, correct? Looks good if so.

File: charts/lily/templates/worker/app.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          {{- toYaml .Values.cluster.worker.resources | nindent 10 }}
+        {{- else }}
+          # Required resources to run daemon lily are signficant and the below default
+          # may be insufficient. Please see https://lilium.sh/software/lily/hardware/
+          # for more information.
+          requests:
+            cpu: "16000m"
+            memory: "250Gi"
+          limits:
+            cpu: "16000m"
```
Comment (frrist) [inline charts/lily/templates/worker/app.yaml:81]: Right now the workers are using at most ~30% of their CPU. How does this differ from our current deployment?
Screen Shot 2022-06-17 at 17 44 07

**Review by frrist — COMMENTED** [review]
@placer14 could you leave some details on what is changing wrt Memory and CPU on the worker and notifier deployments?

File: charts/lily/templates/daemon/app.yaml (hunk 2 above)
Comment (placer14) [reply charts/lily/templates/daemon/app.yaml]: I've refactored the configurations into reused templates in https://github.com/filecoin-project/helm-charts/pull/160/commits/16c350388d7cbc8e53733c825fcf28177d673014

File: charts/lily/templates/worker/app.yaml (hunk 4 above)
Comment (placer14) [reply charts/lily/templates/worker/app.yaml]: The defaults I'm setting in the helm chart are:

daemon and worker:
```
+    resources:
+      requests:
+        cpu: "31000m"
+        memory: "245Gi"
+      limits:
+        cpu: "31000m"
+        memory: "245Gi"
```

notifier:
```
+    resources:
+      requests:
+        cpu: "8000m"
+        memory: "64Gi"
+      limits:
+        cpu: "8000m"
+        memory: "64Gi"
```

These values are not necessarily used for our deployments and are just defaults for the chart. We will tune the resources in our deployments independent of these values.

Comment (placer14) [reply charts/lily/templates/worker/app.yaml]: See https://github.com/filecoin-project/helm-charts/pull/160/commits/03d6241a24283fa385e01206d56f7f4d795839d5

**Pull request merged**